internal/page/donor/how_do_you_know_your_certificate_provider.go

```diff
@@ -40,14 +40,12 @@ func HowDoYouKnowYourCertificateProvider(tmpl template.Template, donorStore Dono
 					requireLength = true
 				}
 
-				if requireLength {
-					// TODO: should stay as Completed if editing and not changing the answer here
-					lpa.Tasks.CertificateProvider = actor.TaskInProgress
-				} else {
-					lpa.CertificateProvider.RelationshipLength = ""
-					lpa.Tasks.CertificateProvider = actor.TaskCompleted
+				if !requireLength {
+					lpa.CertificateProvider.RelationshipLength = 0
 				}
 
+				lpa.Tasks.CertificateProvider = actor.TaskInProgress
```

Comment:
I've just noticed we don't set this in `CertificateProviderDetails` which is where it should be

Reply:
Good spot, I've added that in to the details handler now.